sdk/include/opentelemetry/sdk/logs/logger_provider.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "opentelemetry/logs/logger_provider.h"
+#include "opentelemetry/logs/noop.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/sdk/logs/logger.h"
+#include "opentelemetry/sdk/logs/processor.h"
+
+// Define the maximum number of loggers that are allowed to be registered to the loggerprovider.
+// TODO: Add link to logging spec once this is added to it
+#define MAX_LOGGER_COUNT 100
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+class Logger;
+
+class LoggerProvider final : public opentelemetry::logs::LoggerProvider,
```

Comment:
I remember we talked about make `LoggerProvider` a singleton, or do we need multiple log provider support?

Reply:
I believe a singleton, like traces

Reply:
I believe it was a singleton, like traces - but correct me if I'm wrong.

Reply:
I don't think trace provider should be a singleton.

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#tracerprovider

_Notwithstanding any global TracerProvider, some applications may want to or have to use multiple TracerProvider instances, e.g. to have different configuration (like SpanProcessors) for each (and consequently for the Tracers obtained from them), or because its easier with dependency injection frameworks. Thus, implementations of TracerProvider SHOULD allow creating an arbitrary number of TracerProvider instances._

Reply:
Oh I see- since the spec requires support for multiple TracerProviders, but traces doesn't have support this either yet, should we flag it an issue to update the implementation for traces? So that the specification and implementation is aligned for GA? Or should we implement it for logging first? If so would this PR be a good place to do that, or should we leave it as a later enhancement?

Reply:
I think if you leave the PR as it is, it already has support for multiple providers.

Reply:
Oh I thought you meant to implement GetLoggerProvider() similar to GetLogger() (e.g. using a map to store them all). 

Is there anything else that is needed for this PR? Or is this ready to be merged?

Reply:
Looks good for now and we're good to merge. Would you rebase to the lastest master branch? Looks like I cannot do it due to some issue with GitHub:

![image](https://user-images.githubusercontent.com/17327289/99297506-32f41c00-27fd-11eb-9a64-dd43ed0c8d76.png)